app/feature/insurances/src/main/kotlin/com/hedvig/android/feature/insurances/insurance/presentation/InsurancePresenter.kt

```diff
@@ -145,23 +68,96 @@ internal data class InsuranceUiState(
   }
 }
 
+internal class InsurancePresenter(
+  private val getInsuranceContractsUseCase: GetInsuranceContractsUseCase,
+  private val getCrossSellsUseCase: GetCrossSellsUseCase,
+  private val crossSellCardNotificationBadgeService: CrossSellCardNotificationBadgeService,
+) : MoleculePresenter<InsuranceScreenEvent, InsuranceUiState> {
+  @Composable
+  override fun MoleculePresenterScope<InsuranceScreenEvent>.present(
+    seed: InsuranceUiState,
+  ): InsuranceUiState {
+    var insuranceData by remember {
+      mutableStateOf<InsuranceData>(
+        InsuranceData(
+          insuranceCards = seed.insuranceCards,
+          crossSells = seed.crossSells,
+          quantityOfCancelledInsurances = seed.quantityOfCancelledInsurances,
+        ),
+      )
+    }
+    var isLoading by remember { mutableStateOf(true) }
+    var didFailToLoad by remember { mutableStateOf(false) }
+    val retryChannel = remember { RetryChannel() }
+
+    val showNotificationBadge by crossSellCardNotificationBadgeService
+      .showNotification()
+      .collectAsState(seed.showNotificationBadge)
+
+    CollectEvents { event ->
```

Comment:
This is a custom Composable effect, same as LaunchedEffect, but we get the event directly in here, much more convenient this way